docs/.vuepress/components/cdr-doc-html-example-list.vue

```diff
@@ -0,0 +1,242 @@
+
+  export default {
+    name: 'CdrDocHtmlExampleList',
+    props: {
+      backgroundToggle: {
+        type: Boolean,
+        default: true
+      },
+      backgroundColors: {
+        type: Object,
+        default: function() {
+          return {};
+        }
+      },
+      backgroundColor: {
+        type: String,
+        default: 'light'
+      },
+      interactive: {
+        type: Boolean,
+        default: true
+      },
+      label: {
+        type: [String, Boolean],
+        default: false
+      },
+      showExampleLabels: {
+        type: Boolean,
+        default: true
+      }
+    },
+    data: function() {
+      return {
+        exampleCount: 0,
+        backgroundToggleStates: {},
+        instanceId: null,
+        templateSources: {}
+      }
+    },
+    created: function() {
+      let backgroundToggleStates = {}
+      for (const label in this.$slots) {
+        if (this.backgroundColors[label]) {
+          backgroundToggleStates[label] = this.backgroundColors[label];
+        } else {
+          backgroundToggleStates[label] = this.backgroundColor;
+        }
+      }
+      this.backgroundToggleStates = backgroundToggleStates; // Set default background toggle states
+    },
+    mounted: function () {
+      this.instanceId = this._uid;
+      this.exampleCount = Object.keys(this.$slots).length;
+      for (const label in this.$slots) {
+        const mountId = '#cdr-doc-html-example-' + label;
+        const codeNode = this.extractCodeNodeFromVnodeTree(this.$slots[label][0]);
+        const templateSource = this.getStoredTemplateSourceForExample(label, codeNode);
+        this.renderExampleFromTemplate(templateSource, mountId);
+      }
+    },
+    methods: {
+      getStoredTemplateSourceForExample(label, codeNode) {
+        let templateSource = this.templateSources[label]
+        if (!templateSource) {
+          templateSource = this.extractTextFromVnode(codeNode, '');
+          this.templateSources[label] = templateSource;
+        }
+        return templateSource;
+      },
+      extractTextFromVnode(vNode, text) {
+        let newText = text;
+        if (vNode.text) {
+          newText += vNode.text;
+        } else if (vNode.children) {
+          vNode.children.forEach(childNode => {
+            newText += this.extractTextFromVnode(childNode, '');
+          })
+        } else {
+          return '';
+        }
+        return newText;
+      },
+      extractCodeNodeFromVnodeTree(vNode) {
+        let codeNode = false;
+        if (vNode.tag === 'code') {
+          codeNode = vNode;
+        } else if (vNode.children) {
+          vNode.children.some(childNode => {
+            codeNode = this.extractCodeNodeFromVnodeTree(childNode);
+            return codeNode !== false;
+          });
+        }
+
+        return codeNode
+      },
+      renderExampleFromTemplate (template, mountId) {
+        // This method calls the Vue compiler directly to render the VUE template content as HTML
```

Comment:
this is required to convert snippet from string back to vue?

Reply:
That's correct